### PR TITLE
O3-999: Added outpatient app side panel search bar

### PR DESCRIPTION
**Pull request by CynthiaKamau**
## Requirements

- [x] This PR has a title that briefly describes the work done, on ticket O3-999.
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [x] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary

Duplicate of https://github.com/openmrs/openmrs-esm-patient-management/pull/97 .

## Screenshots
AdvancedSearch
Patient Queue Add Patient To List

## Issue

https://issues.openmrs.org/projects/O3/issues/O3-999

## Screenshots

*None.*

## Related Issue

*None.*

## Other

*None.*

### CODE REVIEW

**Comment by brandones** [comment]
@CynthiaKamau Please do not open new PRs for the same change. Doing that gets rid of the history of your work and our conversation about it. This makes it much harder for whoever is reviewing (in this case, me) to keep track of things.

I tried to close this one and re-open the previous one and it says that the entire repository that had the previous change had been deleted. I don't know why you would do that. There is no reason you should ever have to do that.

You can edit your PR title and description in GitHub directly. You can push updates to your code as new commits to the same branch. It looks like you are using the `main` branch of your fork for everything. I would recommend using branches for different issues. Please [learn how to use git](http://o3-dev.docs.openmrs.org/#/getting_started/prerequisites?id=git) before attempting to contribute.

**Comment by CynthiaKamau** [comment]
> @CynthiaKamau Please do not open new PRs for the same change. Doing that gets rid of the history of your work and our conversation about it. This makes it much harder for whoever is reviewing (in this case, me) to keep track of things.
> 
> I tried to close this one and re-open the previous one and it says that the entire repository that had the previous change had been deleted. I don't know why you would do that. There is no reason you should ever have to do that.
> 
> You can edit your PR title and description in GitHub directly. You can push updates to your code as new commits to the same branch. It looks like you are using the `main` branch of your fork for everything. I would recommend using branches for different issues. Please [learn how to use git](http://o3-dev.docs.openmrs.org/#/getting_started/prerequisites?id=git) before attempting to contribute.

I will have to close this pr and open another one with my new branch

**Comment by brandones** [comment]
No, I promise, you do not have to do that. Please learn to use git. Merge your changes into this branch and push it.

File: packages/esm-outpatient-app/src/patient-search/patient-search.component.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import Overlay from '../overlay.component';
+import PatientAdvancedSearchLaunch from './patient-advanced-search.component';
```
Comment (donaldkibet) [inline packages/esm-outpatient-app/src/patient-search/patient-search.component.tsx:4]:
```suggestion
import PatientAdvancedSearch from './patient-advanced-search.component';
```

File: packages/esm-outpatient-app/src/patient-search/patient-search.component.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import PatientSimpleSearch from './patient-simple-search.component';
+
+interface PatientSearchProps {
+  close: () => void;
+}
+
+const PatientSearch: React.FC<PatientSearchProps> = ({ close }) => {
+  const { t } = useTranslation();
+  const [searchMode, setSearchMode] = useState<SearchMode>(SearchMode.simple);
+  const handleSearchMode = (searchMode: SearchMode) => {
```
Comment (donaldkibet) [inline packages/esm-outpatient-app/src/patient-search/patient-search.component.tsx:15]:
```suggestion
  const toggleSearchMode = (searchMode: SearchMode) => {
```

File: packages/esm-outpatient-app/src/patient-search/patient-simple-search.component.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import { Search, Button, Tile } from 'carbon-components-react';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { SearchMode } from '../types';
+import PatientSearchIcon from './patient-search-icon.component';
+import styles from './patient-simple-search.component.scss';
+
+interface PatientSimpleSearchProps {
+  handleAdvanceSearch: (searchMode: SearchMode) => void;
```
Comment (donaldkibet) [inline packages/esm-outpatient-app/src/patient-search/patient-simple-search.component.tsx:9]:
```suggestion
  toggleSearchMode: (searchMode: SearchMode) => void;
```

**Review by donaldkibet — CHANGES_REQUESTED** [review]
@CynthiaKamau please fix the conflicts, few suggestion in general LGTM 🍻

**Pull request merged**